File: Java/AlgorithmsPractice/src/fizzbuzz/FizzBuzz.java
package fizzbuzz;

import java.util.ArrayList;

public class FizzBuzz {

    private int upUntil;

    public FizzBuzz(int upUntil) {
        this.upUntil = upUntil;
    }

    public ArrayList<String> getFizzbuzz() {
        ArrayList<String> output = new ArrayList<String>();
        int index = 0;
        while (index < this.upUntil) {
            String outputItem = "";
            index++;
            if (index % 3 ==0) {
                outputItem += "fizz";
            }
            if (index % 5 == 0) {
                outputItem += "buzz";
            }
            if (outputItem.isEmpty()) {
                outputItem = Integer.toString(index);
            }
            output.add(outputItem);
        }

        return output;
    }

    public int getUpUntil() {
        return upUntil;
    }


}

File: Kotlin/AlgorithmsPractice/src/fizzbuzz/FizzBuzz.kt
package fizzbuzz

class FizzBuzz(val upUntil: Int) {

    fun getFizzBuzz(): MutableList<Any> {
        var fizzBuzzList: MutableList<Any> = mutableListOf();
        var index: Int = 0;
        while (index < this.upUntil) {
            var stringValue: String = "";
            index++;
            if (index % 3 == 0) {
                stringValue += "fizz"
            }
            if (index % 5 == 0) {
                stringValue += "buzz"
            }
            if (stringValue.isNotEmpty()) {
                fizzBuzzList.add(stringValue)
            } else {
                fizzBuzzList.add(index);
            }
        }
        return fizzBuzzList
    }
}